File: tests/fixtures/vulnerable/client_trust.ts
// Vulnerable: client-supplied authority fields
import express from "express";

const app = express();

app.post("/api/score", (req, res) => {
  const score = req.body.score;
  saveScore(req.user.id, score);
  res.json({ status: "ok" });
});

app.put("/api/user/role", (req, res) => {
  const role = req.body.role;
  updateUserRole(req.user.id, role);
  res.json({ status: "ok" });
});

File: tests/fixtures/vulnerable/client_trust.py
"""Vulnerable: server accepts client-supplied authority fields."""
from flask import Flask, request

app = Flask(__name__)


@app.route("/api/score", methods=["POST"])
def update_score():
    """Bad: score from client request body."""
    score = request.json["score"]
    save_score(current_user.id, score)
    return {"status": "ok"}


@app.route("/api/user/role", methods=["PUT"])
def set_role():
    """Bad: role from client."""
    role = request.json.get("role")
    current_user.role = role
    current_user.save()
    return {"status": "ok"}
